Fix groovy code format with linters

Sunnatillo opened the pull request:
This PR fixes  jenkins pipeline code formats files with Groovy linters

lentzi90 commented on jenkins/jobs/bml_integration_tests.groovy:
```diff
@@ -1,124 +1,121 @@
+                                sh './jenkins/scripts/bare_metal_lab/bml_integration_test.sh'
+                            }
+                        }
+                    }
```
```suggestion
                withCredentials([
                    sshUserPrivateKey(credentialsId: 'metal3ci_city_cloud_ssh_keypair', keyFileVariable: 'METAL3_CI_USER_KEY'),
                    usernamePassword(credentialsId: 'metal3-bml-ilo-credentials', usernameVariable: 'BML_ILO_USERNAME', passwordVariable: 'BML_ILO_PASSWORD'),
                    string(credentialsId: 'metal3-clusterctl-github-token', variable: 'GITHUB_TOKEN')
                ]) {
                    timestamps {
                        sh './jenkins/scripts/bare_metal_lab/bml_integration_test.sh'
                    }
```

lentzi90 commented on jenkins/jobs/bml_integration_tests.groovy:
```diff
@@ -1,124 +1,121 @@
+                        sh './jenkins/scripts/dynamic_worker_workflow/fetch_logs.sh'
+                        archiveArtifacts "logs-${env.BUILD_TAG}.tgz"
+                    }
+                }
```
```suggestion
            withCredentials([
                sshUserPrivateKey(credentialsId: 'metal3ci_city_cloud_ssh_keypair', keyFileVariable: 'METAL3_CI_USER_KEY'),
                usernamePassword(credentialsId: 'metal3-bml-ilo-credentials', usernameVariable: 'BML_ILO_USERNAME', passwordVariable: 'BML_ILO_PASSWORD')
            ]) {
                timestamps {
                    sh './jenkins/scripts/dynamic_worker_workflow/fetch_logs.sh'
                    archiveArtifacts "logs-${env.BUILD_TAG}.tgz"
                }
```

lentzi90 reviewed (commented):
Nice cleanup!
I added suggestions for cleaning up nested `withCredentials` also.
The quoting was a bit confusing for me first but it seems consistent. If the string contains variables, then `"` is used, otherwise `'`. Maybe it helps with accidental evaluation, I don't know but it is ok for me.

Sunnatillo commented:
> Nice cleanup! I added suggestions for cleaning up nested `withCredentials` also. The quoting was a bit confusing for me first but it seems consistent. If the string contains variables, then `"` is used, otherwise `'`. Maybe it helps with accidental evaluation, I don't know but it is ok for me.

Thanks for review @lentzi90. I fixed all nested  withCredentials  issue as well.

tuminoid commented:
Added https://github.com/metal3-io/project-infra/issues/1153 to add prow linter job for groovy after we merge this.

adilGhaffarDev reviewed (approved):
/lgtm
